modules/navigation/godot_navigation_server.cpp

```diff
@@ -635,6 +635,99 @@ void GodotNavigationServer::process(real_t p_delta_time) {
 	}
 }
 
+/////////////////////////////////////
+
+gd::PathQueryResult GodotNavigationServer::_query_path(const gd::PathQueryParameters &p_parameters) const {
+	gd::PathQueryResult query_result;
+
+	const NavMap *map = map_owner.get_or_null(p_parameters.map);
+	ERR_FAIL_COND_V(map == nullptr, query_result);
+
+	// run the pathfinding
+
+	if (p_parameters.pathfinding_algorithm == gd::PathfindingAlgorithm::PATHFINDING_ALGORITHM_ASTAR) {
+		query_result.path = map->get_path(p_parameters.start_position, p_parameters.target_position, p_parameters.optimize, p_parameters.navigation_layers);
+	} else {
+		return query_result;
+	}
+
+	// add path postprocessing
+
+	if (!query_result.path.is_empty() && p_parameters.strip_path) {
+		_strip_path(p_parameters, query_result);
+	}
+
+	// add calculated path stats
+
+	if (query_result.path.size() > 1 && p_parameters.calc_path_length) {
+		_calc_path_length(p_parameters, query_result);
+	}
+
+	return query_result;
+}
+
+void GodotNavigationServer::_strip_path(const gd::PathQueryParameters &p_parameters, gd::PathQueryResult &p_query_result) const {
+	Vector<Vector3> stripped_path;
+
+	int path_inx = 0;
+	Vector3 new_direction;
+	Vector3 past_direction;
+	bool has_past_direction = false;
+	float margin_limit = 1.0 - p_parameters.strip_margin;
+	float dotp = 0.0;
+	bool is_straight = false;
+
+	while (path_inx < p_query_result.path.size()) {
+		Vector3 path_pos = p_query_result.path[path_inx];
+
+		if (path_inx + 1 < p_query_result.path.size()) {
+			Vector3 path_pos_next = p_query_result.path[path_inx + 1];
+
+			if (path_pos == path_pos_next) {
+				path_inx += 1;
+				continue;
+			}
+
+			new_direction = (path_pos_next - path_pos).normalized();
+			is_straight = false;
+
+			if (has_past_direction) {
+				dotp = past_direction.dot(new_direction);
+
+				if (dotp > margin_limit) {
+					is_straight = true;
+				}
+			}
+			past_direction = new_direction;
+			has_past_direction = true;
+
+			if (!is_straight) {
+				stripped_path.push_back(path_pos);
+			}
+		} else {
+			stripped_path.push_back(path_pos);
+		}
+
+		path_inx += 1;
+	}
+
+	p_query_result.path = stripped_path;
+}
+
+void GodotNavigationServer::_calc_path_length(const gd::PathQueryParameters &p_parameters, gd::PathQueryResult &p_query_result) const {
```

Comment:
`p_query_result` should be `r_query_result` instead.

Reply:
How? Shouldn't all Godot function parameters be prefixed with `p_`?. `p_query_result` is from the function PathQueryResult parameter, there is no return.